Code/Tools/ProjectManager/Source/ProjectExportController.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ProjectExportController.h>
+#include <ProjectExportWorker.h>
+#include <ProjectButtonWidget.h>
+#include <SettingsInterface.h>
+
+#include <QMessageBox>
+#include <QDebug>
+#include <QDesktopServices>
+#include <QUrl>
+
+namespace O3DE::ProjectManager
+{
+    ProjectExportController::ProjectExportController(const ProjectInfo& projectInfo, ProjectButton* projectButton, QWidget* parent)
+        : QObject()
+        , m_projectInfo(projectInfo)
+        , m_projectButton(projectButton)
+        , m_parent(parent)
+    {
+        m_worker = new ProjectExportWorker(m_projectInfo);
```

Comment:
Is the lifetime of `m_worker` owned by the `m_workerThread` ? Where is it being cleaned up?

Reply:
Yes the m_worker is owned by m_workerThread, based on the constructor for the ProjectExportController. I believe this is a feature of qt based threads where you can move qt objects into specific threads. Based on that, when the ProjectExportController deconstructor is called, the m_workerThread is torn down, including all resources it presently owns.

Reply:
There's also a connect statement in the constructor where m_worker's `deleteLater` method is called when QThread is finishing up.